inference-engine/tests/functional/plugin/shared/src/single_layer_tests/fake_quantize.cpp

```diff
@@ -18,47 +18,133 @@
     size_t levels;
     std::vector<size_t> constShape;
-    std::tie(levels, constShape) = fqParams;
+    std::vector<float> fqDirectArg;
+    std::vector<float> inputArg;
+    std::tie(levels, constShape, fqDirectArg, inputArg) = fqParams;
+    if (inputArg.size() == 3) {
+        inputDataMin = inputArg[0];
+        inputDataMax = inputArg[1];
+        inputDataResolution = inputArg[2];
+    }
     auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
     auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
     auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
 
-    auto fq = std::dynamic_pointer_cast<ngraph::opset1::FakeQuantize>(ngraph::builder::makeFakeQuantize(paramOuts[0], ngPrc, levels, constShape));
+    UpdateSeed();
+
+    std::shared_ptr<ngraph::Node> fakeQNode;
+    if (fqDirectArg.empty()) {
+        int32_t ngraphSeed = seed;
+        if (NGRAPH_SEED != USE_CLOCK_TIME) {
+            ngraphSeed = NGRAPH_SEED;
+        }
+        std::cout << "\033[0;32m" << "[          ] " << "\033[0;0m"
+                  << "ngraphSeed = " << ngraphSeed << std::endl;
+        fakeQNode = ngraph::builder::makeFakeQuantize(paramOuts[0], ngPrc, levels, constShape, ngraphSeed);
+    } else {
+        fakeQNode = ngraph::builder::makeFakeQuantize(
+            paramOuts[0],
+            ngPrc,
+            levels,
+            constShape,
+            {fqDirectArg[0]},
+            {fqDirectArg[1]},
+            {fqDirectArg[2]},
+            {fqDirectArg[3]});
+    }
+
+
+    auto fq = std::dynamic_pointer_cast<ngraph::opset1::FakeQuantize>(fakeQNode);
 
     ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(fq)};
     function = std::make_shared<ngraph::Function>(results, params, "fakeQuantize");
+
+    configuration = config.second;
+}
+
+InferenceEngine::Blob::Ptr FakeQuantizeLayerTest::GenerateInput(const InferenceEngine::InputInfo &info) const {
+    return FuncTestUtils::createAndFillBlob(info.getTensorDesc(), inputDataMax - inputDataMin, inputDataMin, 1 / inputDataResolution, seed);
+}
+
+void FakeQuantizeLayerTest::UpdateSeed() {
+    if (BASE_SEED == USE_CLOCK_TIME) {
+        seed = std::chrono::system_clock::now().time_since_epoch().count();
+    } else if (BASE_SEED == USE_INCREMENTAL_SEED) {
+        seed += 9999;
+    } else {
+        seed = BASE_SEED;
+    }
+    std::cout << "\033[0;32m" << "[          ] " << "\033[0;0m"
+              << "seed = " << seed << std::endl;
 }
 
 TEST_P(FakeQuantizeLayerTest, CompareWithRefs) {
     Run();
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+
+    if (BASE_SEED != USE_CLOCK_TIME &&
+        BASE_SEED != USE_INCREMENTAL_SEED) {
+        return;
+    }
+
+    size_t nIterations = (inputDataMax - inputDataMin) / inputDataResolution;
+    for (; nIterations != 0; nIterations--) {
+        UpdateSeed();
+        Infer();
+        Validate();
+    }
```

Comment:
every infer proceed with different input from given range min, min+step, min+2*step, ... max  that is why results are different and whole range a bit covered.

Reply:
So you say that smaller step is, the more input vectors you would want to check because there are more possible values? why not make sure that test actually tests every single possible input? number of computations would be even less when issue: 38806 is solved and 4d case is uncommented. But it can stay as is, at least i know the logic behind.

Reply:
every possible value is too many - but possible like if you will increment using std::nextafter, in that case will test like i dont know more then milion combinations in our interest of interval this will dramatically increase testing time. My observations showed that for example x.5 values that are sensitive for rounding, and x.499999 values  that are sensitive for mul/div order are easily hit using current step. 
More values are subject to stress testing.

Reply:
i meant input size  {3, 10, 5, 6} have  900 values, for step 0.1 it means 900*10 = 9000 calculations, while input reange and step {-10, 10, 0.1} leads to just 200 possible values. So for 4d case 1 iteration could test all possible values 4.5 times. I do not force you to change the code, it is just an observation, that for this test and these particular parameters used it is possible to test all values that can potentially be generated by GenerateInput() with less number of calculations (but only after 4d case is re-enabled).

Reply:
4d case not supported yet as you can see, and i think we can add different threshold there.